### PR TITLE
feat: Expose with grpc

### PR DESCRIPTION
**Pull request by prakash100198**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 RUN apk add --update make
 RUN mkdir /silver-surfer
 WORKDIR /silver-surfer
 ADD . /silver-surfer/
-ARG AUTH_TOKEN
-RUN test -n "$AUTH_TOKEN"
-ENV GITHUB_TOKEN=${AUTH_TOKEN}
-ARG RELEASE
-RUN if [ "$RELEASE" = "goreleaser" ]; then echo `make release`; fi
+#ARG AUTH_TOKEN
```
Comment (nishant-d) [inline Dockerfile:9]: uncomment this

File: app/App.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	// Start gRPC server
+	err = app.initGrpcServer(app.StartupConfig.GrpcPort)
+	if err != nil {
+		app.logger.Errorw("error starting grpc server", "err", err)
+		os.Exit(2)
+	}
+}
+
+func (app *App) initGrpcServer(port int) error {
```
Comment (nishant-d) [inline app/App.go:62]: add prometheus

File: main.go [hunk 3]
```diff
@@ -285,5 +288,25 @@ func init() {
 }
 
 func main() {
-	Execute()
+	useAsCli := os.Getenv("USE_AS_CLI")
+	if useAsCli == "true" {
+		Execute()
+	} else {
+		app, err := InitializeApp()
```
Comment (nishant-d) [inline main.go:295]: create separate main file

File: pkg/Cluster.go [hunk 4]
```diff
@@ -62,6 +66,45 @@ func NewCluster(kubeconfig string, kubecontext string) *Cluster {
...
+			panic(err)
+		}
+		kubeconfig := flag.String("kubeconfig", filepath.Join(usr.HomeDir, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+		//flag.Parse()
+		restConfig, err = clientcmd.BuildConfigFromFlags("", *kubeconfig)
+		if err != nil {
+			panic(err)
+		}
+	} else {
+		restConfig, err = rest.InClusterConfig()
```
Comment (nishant-d) [inline pkg/Cluster.go:86]: add third case

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: app/App.go (hunk 2 above)
Comment (prakash100198) [reply app/App.go:63]: done

File: main.go (hunk 3 above)
Comment (prakash100198) [reply main.go]: done

File: pkg/Cluster.go (hunk 4 above)
Comment (prakash100198) [reply pkg/Cluster.go]: done

File: Dockerfile (hunk 1 above)
Comment (prakash100198) [reply Dockerfile]: done

File: pkg/Cluster.go [hunk 5]
```diff
@@ -62,6 +66,47 @@ func NewCluster(kubeconfig string, kubecontext string) *Cluster {
 	return &cluster
 }
 
+func NewClusterViaInClusterConfig(restConfig *rest.Config) *Cluster {
```
Comment (nishant-d) [inline pkg/Cluster.go:69]: name it properly

File: DockerfileMVC
Comment (nishant-d) [inline DockerfileMVC:1]: move it inside app folder

**Review by nishant-d — CHANGES_REQUESTED** [review]

**Pull request merged**